maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveProtobufSchema.java

```diff
@@ -0,0 +1,143 @@
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
+
+import java.util.List;
+
+import static io.confluent.kafka.schemaregistry.maven.derive.schema.DeriveSchemaUtils.getSortedKeys;
+
+public class DeriveProtobufSchema extends DeriveSchema {
+
+  public static final String DOUBLE = "double";
+  public static final String STRING = "string";
+  public static final String BOOL = "bool";
+  public static final String INT_32 = "int32";
+  public static final String INT_64 = "int64";
+  public static final String ANY_FIELD = "google.protobuf.Any";
+  public static final String FIELD_ENTRY = " %s %s = %d;%n";
+  public static final String PROTOBUF_SYNTAX = "syntax = \"proto3\";\n";
+
+  public DeriveProtobufSchema() {
+    // Map jackson node data type to type understood by protobuf
+    classToDataType.put(com.fasterxml.jackson.databind.node.DoubleNode.class.getName(), DOUBLE);
+    classToDataType.put(com.fasterxml.jackson.databind.node.TextNode.class.getName(), STRING);
+    classToDataType.put(com.fasterxml.jackson.databind.node.BigIntegerNode.class.getName(), DOUBLE);
+    classToDataType.put(com.fasterxml.jackson.databind.node.IntNode.class.getName(), INT_32);
+    classToDataType.put(com.fasterxml.jackson.databind.node.LongNode.class.getName(), INT_64);
+    classToDataType.put(com.fasterxml.jackson.databind.node.BooleanNode.class.getName(), BOOL);
+    classToDataType.put(com.fasterxml.jackson.databind.node.NullNode.class.getName(), ANY_FIELD);
+    classToDataType.put(com.fasterxml.jackson.databind.node.MissingNode.class.getName(), ANY_FIELD);
+  }
+
+  protected ObjectNode mergeMultipleDataTypes(ObjectNode mergedArray,
+                                              List<JsonNode> primitives,
+                                              List<JsonNode> records,
+                                              List<JsonNode> arrays,
+                                              boolean check2dArray) {
+    ArrayNode items = mapper.createArrayNode();
+    // Check for nested array, if yes return error
+    if (arrays.size() > 0 && check2dArray) {
+      throw new IllegalArgumentException(String.format("found nested array: %s", arrays));
+    }
+
+    DeriveSchemaUtils.mergeNumberTypes(primitives);
+    // Adding primitive types to items' list
+    items.addAll(DeriveSchemaUtils.getUnique(primitives));
+    // No merging of arrays, add directly to items' list
+    items.addAll(arrays);
+
+    // Merge records if there is at least 1 record
+    if (records.size() > 0) {
+      items.add(mergeRecords(records));
+    }
+
+    if (items.size() > 1) {
+      throw new IllegalArgumentException(String.format("found multiple data types: %s", items));
+    } else if (items.size() == 1) {
+      mergedArray.set("items", items.get(0));
+    } else {
+      mergedArray.set("items", getNullSchema());
+    }
+    return mergedArray;
+  }
+
+  @Override
+  public TextNode convertToFormat(JsonNode schema, String name) {
```

Comment:
this is public here but protected in the base class and also DeriveJsonSchema as well. why is that?